pkgs/by-name/as/asmc/package.nix

```diff
@@ -0,0 +1,61 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "asmc";
+  version = "2.36.02-unstable-2024-11-01";
+
+  src = fetchFromGitHub {
+    owner = "nidud";
+    repo = "asmc";
+    rev = "7dda9a180031c3723e270e0b6f51f4d8b73004a9";
+    hash = "sha256-Uk9K8m8r9stetIlUoVfBlayeDZSDvt1EmTI68W8dwso=";
+  };
+
+  makeFlags = [
+    "-C"
+    "source/asmc"
+    # Fails to build with the PIE default, sadly.
+    "pic-default=no"
+  ];
+
+  hardeningDisable = [ "pie" ];
+
+  # Lots of undeclared dependencies.
+  enableParallelBuilding = false;
+
+  postPatch = ''
+    # Unconditionally passes `-Wl,-pie` even when PIC is disabled, and
+    # then fails to build with it.
+    substituteInPlace source/asmc/makefile \
+      --replace-fail 'gcc -Wl,-pie' gcc
+  '';
+
+  # `make install` hard‐codes `/usr` and `sudo`; not worth it.
+  #
+  # This doesn’t handle the include or library directories because we
+  # just use this for `_7zz` and don’t install any of the libraries.
+  # Better to fail fast if anyone needs them so they know they’ll need
+  # to adjust this derivation.
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin
+    cp source/asmc/{asmc,asmc64} $out/bin
+
+    runHook postInstall
+  '';
+
+  meta = {
+    description = "MASM‐compatible assembler";
+    homepage = "https://github.com/nidud/asmc";
+    changelog = "https://github.com/nidud/asmc/blob/${finalAttrs.src.rev}/source/asmc/history.txt";
+    license = lib.licenses.gpl2Only;
+    sourceProvenance = [ lib.sourceTypes.fromSource ];
```

Comment:
I did a bit of reading and these details might not be right :cry: 

https://github.com/nidud/asmc/issues/15

> Asmc is a fork of [JWasm](https://github.com/Baron-von-Riedesel/JWasm) so it was originally written in C.

With JWasm using the Open Watcom license we're having problems with

https://github.com/Baron-von-Riedesel/JWasm/blob/036df3a2e7d5b7d9ecb1e162507d93b1f92ec285/Html/License.html#L13

https://github.com/Terraspace/UASM/blob/master/License.txt#L1-L12

I don't know the ins-and-outs of the "fork" but IDK if asmc can count as just gpl2Only :thinking:
I'm not giving specific statements here, just questioning.

---

> > It's weird to have an open source project that relies on binary blobs.
> All open source relies on binary blobs.

Looks like it was changed to not use a binary for the build but does rely on a good chunk of assembly

> I've managed to produce a somewhat functional source for the GNU assembler.

> Okey, so I will just use the modified source then.
> The Linux binaries are now removed and the makefile updated.

https://github.com/nidud/asmc/commit/2327fb98340f75c39d6809409a35feb7509a012a

It's not exactly bytecode but would this be a case to use `binaryBytecode` or from source is fine?